packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+import { catchError, concatMap, filter, first, map, observeOn, switchMap, tap, withLatestFrom } from 'rxjs/operators';
+
+import { fromAdapterEvent, isNotNull } from '../operators';
+import { WalletNotSelectedError } from './wallet.errors';
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+export const WALLET_DEFAULT_CONFIG: WalletConfig = {
+    wallets: [],
+    autoConnect: false,
+    localStorageKey: 'walletName',
+    onError: (error: unknown) => console.error(error),
+};
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
+    private readonly _localStorageKey = this._config?.localStorageKey || 'walletName';
+    private logError = this._config?.onError || ((error: unknown) => console.error(error));
+    readonly wallets$ = this.select((state) => state.wallets);
+    readonly selectedWallet$ = this.select((state) => state.selectedWallet);
+    readonly connected$ = this.select((state) => state.connected);
+    readonly connecting$ = this.select((state) => state.connecting);
+    readonly disconnecting$ = this.select((state) => state.disconnecting);
+    readonly wallet$ = this.select((state) => state.wallet);
+    readonly adapter$ = this.select((state) => state.adapter);
+    readonly publicKey$ = this.select((state) => state.publicKey);
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(
+        this.publicKey$.pipe(isNotNull),
+        this.adapter$.pipe(isNotNull),
+        (publicKey, adapter) => {
+            return publicKey && 'signTransaction' in adapter && 'signAllTransactions' in adapter
+                ? {
+                      publicKey,
+                      signTransaction: (transaction: Transaction) => this.signTransaction(transaction)?.toPromise(),
+                      signAllTransactions: (transactions: Transaction[]) =>
+                          this.signAllTransactions(transactions)?.toPromise(),
+                  }
+                : undefined;
+        }
+    );
+
+    constructor(
+        @Optional()
+        @Inject(WALLET_CONFIG)
+        private _config: WalletConfig
+    ) {
+        super({
+            wallets: _config.wallets,
+            selectedWallet: null,
+            wallet: null,
+            adapter: null,
+            connected: false,
+            connecting: false,
+            disconnecting: false,
+            ready: false,
+            publicKey: null,
+            autoApprove: false,
+        });
+
+        if (!this._config) {
+            this._config = WALLET_DEFAULT_CONFIG;
+        } else {
+            this._config = {
+                ...WALLET_DEFAULT_CONFIG,
+                ...this._config,
+            };
+        }
+
+        const walletName = localStorage.getItem(this._localStorageKey);
```

Comment:
In the react package we JSON encode this value, so the literal value is e.g. `"WalletName"`.

Reply:
Would you mind elaborating on this? I think don't get it

Reply:
LocalStorage values need to be strings, so to encode any kind of data, it's a common pattern to use a wrapper around the API to `JSON.stringify` in and then `JSON.parse` out, which has the effect of quoting strings.

This probably doesn't matter (it's not like you're just going to switch your app from using React to Angular and care if wallets autoConnect) but I'm thinking we should do it for consistency.

Reply:
Ohh I see, my train of thought was: If the walletName is a string, I don't need to encode/decode. But I just noticed in the React package it's a WalletName | null, not a plain string. I think I got it.

Reply:
Ah, WalletName is an enum of a string without the quotes. But if you look at the `useLocalStorage` hook, that's where the wrap/unwrap and typecasting happens.